src/PantheonTerminalWindow.vala

```diff
@@ -1086,13 +1079,16 @@ namespace PantheonTerminal {
                     {"Escape", "<Ctrl><Shift>f"},
                     _("Hide find bar")
                 );
+                search_toolbar.grab_focus ();
             } else {
                 action_accelerators.remove_all(ACTION_SEARCH_NEXT);
                 action_accelerators.remove_all(ACTION_SEARCH_PREVIOUS);
                 search_button.tooltip_markup = Granite.markup_accel_tooltip (
                     {"<Ctrl><Shift>f"},
                     _("Find…")
                 );
+                search_toolbar.clear ();
+                current_terminal.grab_focus ();
             }
             app.set_accels_for_action (ACTION_PREFIX + ACTION_SEARCH_NEXT,
```

Comment:
This results in a critical terminal warning when the accelerator collection is empty. It seems a null array is returned by to_array () instead of an empty one?  This may be an upstream bug.  A work around is:

```
var accels = action_accelerators[ACTION_SEARCH_NEXT].is_empty ? new string [] {} : action_accelerators[ACTION_SEARCH_NEXT].to_array ();
            app.set_accels_for_action (ACTION_PREFIX + ACTION_SEARCH_NEXT,
                                       accels);

```